src/main/java/application/models/MultipleChoiceQuestion.java

```diff
@@ -1,9 +1,105 @@
 package application.models;
 
 import javax.persistence.DiscriminatorValue;
+import javax.persistence.ElementCollection;
 import javax.persistence.Entity;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
 
+/**
+ * Subclass of Question class
+ * Create a multiple choice question
+ */
 @Entity
 @DiscriminatorValue("1")
 public class MultipleChoiceQuestion extends Question{
+
+    /**
+     *  Holds the collection of choices a user can choose from
+     */
+    @ElementCollection
+     private Collection<String> choices = new ArrayList<String>();
+
+
+
+    /**
+     * Stores a mapping of choices and the number of times that choice was picked
+     */
+    @ElementCollection
+    private Map<String,Integer> answers = new HashMap<String,Integer>();
```

Comment:
I guess another approach is to store choices as a Map, with an integer id as key and choice text as value.
Then make the answers as map (k:id, v:count). This will save HashCode computation time if someone has some insane long choice text as a way to do DOS, which is probably not practical anyway :P

Correspondingly the view can do radio button with the numerical id as the value. And the controller using the API can do **_addAnswer(int (1))_** instead of **_addAnswer(String “Horatio says 'tis but our fantasy, And will not let belief take hold……”)_**

Just my 2 cents about string as the key in a map.

Reply:
You make a good point about having the question as the key. Another possible way would be to just have a list that holds every answer inputted. You could then count the similar strings in the list to find out the results

Reply:
@cyang030 Yes that makes sense, I'm just unsure how to show the results later with their respective choice. If we use (k:id, v:count) would we then need to go back to view and get the string value for each id in the mapping to display the results.

@thomas7carriere I think a mapping would be better than a list of results, if we have for example 500 responses then we would have a list of that long, so we can just update the total results each time its answered instead.

Reply:
@Dom123k Ya, my thinking was to keep things consistent between the other 2 question classes.